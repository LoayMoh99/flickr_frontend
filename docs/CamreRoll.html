<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Namespace: CamreRoll</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Namespace: CamreRoll</h1>

    




<section>

<header>
    
        <h2>CamreRoll</h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        
            <div class="description"><p>Renders CameraRoll photos</p></div>
        

        


<dl class="details">

    

    

    

    

    

    

    

    

    
    <dt class="tag-author">Author:</dt>
    <dd class="tag-author">
        <ul>
            <li>Khadija Khaled</li>
        </ul>
    </dd>
    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="CameraRoll_CamreRoll.jsx.html">CameraRoll/CamreRoll.jsx</a>, <a href="CameraRoll_CamreRoll.jsx.html#line71">line 71</a>
    </li></ul></dd>
    

    

    

    
</dl>


        
    
    </div>

    
        <h3 class="subsection-title">Extends</h3>

        


    <ul>
        <li>Component</li>
    </ul>


    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id=".closeMainModal"><span class="type-signature">(static) </span>closeMainModal<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>To close Main modal , clear count of selected photos and clear the Edit array</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="CameraRoll_CamreRoll.jsx.html">CameraRoll/CamreRoll.jsx</a>, <a href="CameraRoll_CamreRoll.jsx.html#line143">line 143</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id=".containsObject"><span class="type-signature">(static) </span>containsObject<span class="signature">(obj)</span><span class="type-signature"> &rarr; {boolean}</span></h4>
    

    



<div class="description">
    <p>To check if it is present in the array of photos selected or not</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>obj</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last"><p>A photo object</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="CameraRoll_CamreRoll.jsx.html">CameraRoll/CamreRoll.jsx</a>, <a href="CameraRoll_CamreRoll.jsx.html#line33">line 33</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <ul>
<li>A boolean indicating the photo id is present within the array or not</li>
</ul>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">boolean</span>


    </dd>
</dl>

    





        
            

    

    
    <h4 class="name" id=".handleDecrement"><span class="type-signature">(static) </span>handleDecrement<span class="signature">(c)</span><span class="type-signature"> &rarr; {integer}</span></h4>
    

    



<div class="description">
    <p>To decrements count</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>c</code></td>
            

            <td class="type">
            
                
<span class="param-type">integer</span>


            
            </td>

            

            

            <td class="description last"><p>count</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="CameraRoll_CamreRoll.jsx.html">CameraRoll/CamreRoll.jsx</a>, <a href="CameraRoll_CamreRoll.jsx.html#line60">line 60</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <ul>
<li>count decremented</li>
</ul>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">integer</span>


    </dd>
</dl>

    





        
            

    

    
    <h4 class="name" id=".handleDelete"><span class="type-signature">(static) </span>handleDelete<span class="signature">(list, obj)</span><span class="type-signature"> &rarr; {array}</span></h4>
    

    



<div class="description">
    <p>To delete objectt from list</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>list</code></td>
            

            <td class="type">
            
                
<span class="param-type">array</span>


            
            </td>

            

            

            <td class="description last"><p>array of photos</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>obj</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last"><p>A photo object</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="CameraRoll_CamreRoll.jsx.html">CameraRoll/CamreRoll.jsx</a>, <a href="CameraRoll_CamreRoll.jsx.html#line45">line 45</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>-array of photos after deletion</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">array</span>


    </dd>
</dl>

    





        
            

    

    
    <h4 class="name" id=".handleIncrement"><span class="type-signature">(static) </span>handleIncrement<span class="signature">(c)</span><span class="type-signature"> &rarr; {integer}</span></h4>
    

    



<div class="description">
    <p>To increments count</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>c</code></td>
            

            <td class="type">
            
                
<span class="param-type">integer</span>


            
            </td>

            

            

            <td class="description last"><p>count</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="CameraRoll_CamreRoll.jsx.html">CameraRoll/CamreRoll.jsx</a>, <a href="CameraRoll_CamreRoll.jsx.html#line22">line 22</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <ul>
<li>count incremented</li>
</ul>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">integer</span>


    </dd>
</dl>

    





        
            

    

    
    <h4 class="name" id=".monthName"><span class="type-signature">(static) </span>monthName<span class="signature">(item)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>To change date format</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>item</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last"><p>photo object toe change its createdAt date format</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="CameraRoll_CamreRoll.jsx.html">CameraRoll/CamreRoll.jsx</a>, <a href="CameraRoll_CamreRoll.jsx.html#line154">line 154</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id=".toggleAddModal"><span class="type-signature">(static) </span>toggleAddModal<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>To open/close Add modal</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="CameraRoll_CamreRoll.jsx.html">CameraRoll/CamreRoll.jsx</a>, <a href="CameraRoll_CamreRoll.jsx.html#line106">line 106</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id=".toggleAEditModal"><span class="type-signature">(static) </span>toggleAEditModal<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>To open/close Edit modal</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="CameraRoll_CamreRoll.jsx.html">CameraRoll/CamreRoll.jsx</a>, <a href="CameraRoll_CamreRoll.jsx.html#line134">line 134</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id=".toggleCreateAlbum"><span class="type-signature">(static) </span>toggleCreateAlbum<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>To open/close Create Album modal</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="CameraRoll_CamreRoll.jsx.html">CameraRoll/CamreRoll.jsx</a>, <a href="CameraRoll_CamreRoll.jsx.html#line115">line 115</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id=".toggleDelete"><span class="type-signature">(static) </span>toggleDelete<span class="signature">(id)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>To open/close Delete modal</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>id</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>photo id to delete</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="CameraRoll_CamreRoll.jsx.html">CameraRoll/CamreRoll.jsx</a>, <a href="CameraRoll_CamreRoll.jsx.html#line124">line 124</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id=".toggleModal"><span class="type-signature">(static) </span>toggleModal<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>To toggle the main modal open/close based on photo selection/unselection</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="CameraRoll_CamreRoll.jsx.html">CameraRoll/CamreRoll.jsx</a>, <a href="CameraRoll_CamreRoll.jsx.html#line163">line 163</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Namespaces</h3><ul><li><a href="AddAlbumImages.html">AddAlbumImages</a></li><li><a href="AddModal.html">AddModal</a></li><li><a href="AddPhotos.html">AddPhotos</a></li><li><a href="AlbumCard.html">AlbumCard</a></li><li><a href="AlbumCoverPhoto.html">AlbumCoverPhoto</a></li><li><a href="AlbumPage.html">AlbumPage</a></li><li><a href="AlbumPreview.html">AlbumPreview</a></li><li><a href="CamreRoll.html">CamreRoll</a></li><li><a href="Card.html">Card</a></li><li><a href="CommentBox.html">CommentBox</a></li><li><a href="CreateNewAlbumModal.html">CreateNewAlbumModal</a></li><li><a href="DropdownPrivacy.html">DropdownPrivacy</a></li><li><a href="EditInfo.html">EditInfo</a></li><li><a href="EditModal.html">EditModal</a></li><li><a href="Follow.html">Follow</a></li><li><a href="Followers.html">Followers</a></li><li><a href="FollwingFollowers.html">FollwingFollowers</a></li><li><a href="GroupCard.html">GroupCard</a></li><li><a href="GroupHeader.html">GroupHeader</a></li><li><a href="GroupPhotos.html">GroupPhotos</a></li><li><a href="ImageDetails.html">ImageDetails</a></li><li><a href="ImageGrid.html">ImageGrid</a></li><li><a href="ImagesCR.html">ImagesCR</a></li><li><a href="ImagesMM.html">ImagesMM</a></li><li><a href="Join.html">Join</a></li><li><a href="Modal.html">Modal</a></li><li><a href="NavBar.html">NavBar</a></li><li><a href="PeopleCard.html">PeopleCard</a></li><li><a href="PhotosToAdd.html">PhotosToAdd</a></li><li><a href="Photostream.html">Photostream</a></li><li><a href="SearchGroup.html">SearchGroup</a></li><li><a href="SearchPeople.html">SearchPeople</a></li><li><a href="SideNavBar.html">SideNavBar</a></li><li><a href="Upload.html">Upload</a></li><li><a href="UserInfo.html">UserInfo</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.7</a> on Tue Jun 08 2021 19:30:23 GMT+0200 (Eastern European Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>