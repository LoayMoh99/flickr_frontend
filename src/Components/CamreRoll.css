#topNavbar {
  border-style: none;
}
.container_body {
  padding: 1px 170px 10px 170px;
}
main {
  background: #f3f5f6;
}
