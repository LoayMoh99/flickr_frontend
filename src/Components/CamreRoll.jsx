/* eslint-disable linebreak-style */
/* eslint-disable react/style-prop-object */
/* eslint-disable linebreak-style */
import React from 'react';
import ImagesCR from './ImagesCR';

function CamreRoll() {
  const images = [{ Url: 'https://picsum.photos/id/237/200/300', dateupdated: '20-4-2021' }, { Url: 'https://picsum.photos/seed/picsum/200/300', dateupdated: '20-4-2021' }];

  return (

    <>

      <ul className="nav nav-tabs">
        <span className="col" />
        <div className="col-11">
          <li className="nav-item dropdown">
            <a className="nav-link dropdown-toggle text-secondary" data-bs-toggle="dropdown" href="/#" role="button" aria-expanded="false">Date uploaded</a>
            <div className="dropdown-menu">
              <ul>
                <a className="dropdown-item" href="/#">Date uploaded</a>
                <a className="dropdown-item" href="/#">Date taken</a>
              </ul>
            </div>
          </li>

        </div>
      </ul>
      <div className="row">
        <div className="col">
          <nav className="nav flex-column">
            <a className="nav-link active " aria-current="page" href="/#">2021</a>
            <a className="nav-link " href="/#">- April</a>
            <a className="nav-link text-secondary" href="/#">- January</a>
          </nav>
        </div>
        <div className="col-11">
          <div className="row">
            { images.map((image) => (
              <ImagesCR
                Url={image.Url}
                dateupdated={image.dateupdated}
              />
            )) }
          </div>
        </div>
      </div>
    </>
  );
}

export default CamreRoll;
