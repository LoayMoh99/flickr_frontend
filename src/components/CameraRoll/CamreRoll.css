#topNavbar {
  border-style: none;
}
.container_body {
  padding: 1px 10px 10px 75px;
}
.main_Modal {
  background: #f3f5f6;
}
