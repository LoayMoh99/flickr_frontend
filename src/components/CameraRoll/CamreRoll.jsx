/* eslint-disable linebreak-style */
/* eslint-disable no-undef */
/* eslint-disable linebreak-style */
/* eslint-disable max-len */
/* eslint-disable react/jsx-indent */
/* eslint-disable linebreak-style */
/* eslint-disable no-use-before-define */
/* eslint-disable linebreak-style */
/* eslint-disable brace-style */
/* eslint-disable linebreak-style */
/* eslint-disable no-console */
/* eslint-disable linebreak-style */
/* eslint-disable react/no-this-in-sfc */
/* eslint-disable linebreak-style */
/* eslint-disable no-unused-vars */
/* eslint-disable linebreak-style */
/* eslint-disable react/style-prop-object */
/* eslint-disable linebreak-style */
import React, { useState , useEffect } from 'react';
import _, { filter, map, set } from 'lodash';
import moment from 'moment';
import ImagesCR from './ImagesCR';
import Modal from './Modal';
import EditModal from './EditModal';
import AddModal from './AddModal';
import SideNavBar from './SideNavBar';
import DeleteModal from '../DeleteModal/DeleteModal'
import './CamreRoll.css';
import './EditModal.css';
import './AddModal.css';
// import { mockComponent } from 'react-dom/test-utils';
import axios from "axios"
import CreateNewAlbumModal from './CreateNewAlbumModal';
const endpoint = 'http://localhost:3001/'

function CamreRoll() {
  // const images = [{ photo_url: 'https://picsum.photos/id/237/200/300', createdAt: new Date('2019-05-28'), photo_id: '1' }, { photo_url: 'https://picsum.photos/200', createdAt: new Date('2019-06-10'), photo_id: '2' }, { photo_url: 'https://picsum.photos/seed/picsum/200/300', createdAt: new Date('2019-06-11'), photo_id: '3' }, { photo_url: 'https://picsum.photos/200/300?grayscale', createdAt: new Date('2019-06-10'), photo_id: '4' }, { photo_url: 'https://picsum.photos/seed/picsum/200/300', createdAt: new Date('2019-06-10'), photo_id: '5' }, { photo_url: 'https://picsum.photos/seed/picsum/200/300', createdAt: new Date('2019-06-10'), photo_id: '6' }];
  
  //Get photos
  const [images, setImages] = useState([]);
  useEffect(() => {
      const fetchData = async () => {
      const {data,status} = await axios.get( endpoint+'photos',);
      console.log(status);
      if (status === 200){
          setImages(data);
      }
  };
  
    fetchData();
  },[]);
  
  const sortedImagesUploaded = images.slice().sort((a, b) => b.createdAt - a.createdAt);
  const [isModalOpen, setModalIsOpen] = useState(false);
  const [count, setCount] = useState(0);
  const [toEdit, setToEdit] = useState([]);
  const [isEditModalOpen, setEditModalOpen] = useState(false);
  const [isAddModalOpen, setAddModalOpen] = useState(false);
  const [isDeleteOpen, setDeleteOpen] = useState(false);
  const [isCreateAlbumOpen , setCreateAlbumOpen] = useState(false);
  const [idToDelete,setID]=useState(0);
  const [imgDated, setimgDated] = useState([]);
  const [toEditIds, setToEditIds] = useState([]);

  
  const toggleAddModal = () => {
    setAddModalOpen(!isAddModalOpen);
  };
  const toggleCreateAlbum = () => {
    setCreateAlbumOpen(!isCreateAlbumOpen);
  };

  const toggleDelete = (id) => {
    setDeleteOpen(!isDeleteOpen);
    setID(id);
  };

  const toggleEditModal = () => {
    setEditModalOpen(!isEditModalOpen);
  };
  const closeMainModal = () => {
    setModalIsOpen(!isModalOpen);
    setCount(0);
    setToEdit([]);
    // should alse clear the count and clear the to Edit array
  };

  function confirmDelete(){
    if (idToDelete ===0){
        console.log("delete card 1");
    }else if (idToDelete===1){
        console.log("delete card 2");
    } 

    toggleDelete(); 
}

  const monthName = (item) => moment(item.createdAt, 'YYYY-MM-DD').format('DD MMMM YYYY');
  // function to check if this image was already selected or a newly selected one
  function containsObject(obj, list) {
    return list.some((elem) => elem.photo_id === obj.photo_id);
  }
  // to delete the element if unselected
  function handleDelete(obj, list) {
    const listClone = [...list];
    const index = listClone.indexOf(obj);
    // Edit
    return listClone.splice(index, 1);
  }

  function handleIncrement(c) {
    return c + 1;
  }
  function handleDecrement(c) {
    return c - 1;
  }

  // to toggle the modal .. if open then close and vice versa
  const toggleModal = (e, imgObj) => {
    // if count was initially 0 .. this the first image to be selected .. open modal
    let countCopy = count;
    console.log(countCopy);
    if (!countCopy) {

      setToEdit([]);
      setToEditIds([]);

      if (!isModalOpen) {

        setModalIsOpen(!isModalOpen);

        setToEdit((prevItems) => [...prevItems, imgObj]);
        setToEditIds((prevItems)=>[...prevItems,imgObj.photo_id]);


        countCopy = handleIncrement(countCopy);
        setCount(countCopy);
        console.log(countCopy);
        // we now checked if the image was already selected .. unselect it.. count --
        // if no image left close modal
      } else {
        // setModalIsOpen(false);
        setModalIsOpen(!isModalOpen);
      }
    } else {
      const isHere = containsObject(imgObj, toEdit);
      
      console.log(isHere);
      if (!isHere) {

        setToEdit((prevItems) => [...prevItems, imgObj]);
        setToEditIds((prevItems)=>[...prevItems,imgObj.photo_id]);

        countCopy = handleIncrement(countCopy);
        setCount(countCopy);
      }
      console.log(count);
      if (isHere) {

        setToEdit(handleDelete(imgObj, toEdit));
        setToEditIds(handleDelete(imgObj.photo_id, toEditIds));

        countCopy = handleDecrement(countCopy);
        setCount(countCopy);
      }
      console.log(count);
      // we now checked if the image was already selected .. unselect it.. count --
      // if no image left close modal
      console.log(toEdit);
      console.log(toEditIds);

      if (!countCopy) {
     
        setToEdit([]);
        setToEditIds([]);

        setModalIsOpen(!isModalOpen);
      }
    }

        };

      // // extract ids from toEdit photos array
      // toEdit.map(photoSelected=>setToEditIds((prevItems) => [...prevItems,photoSelected.photo_id]));
      //   console.log(toEditIds);

        // function extractToEditIds(){
        //   let prevItems=[];
        //  toEdit.map(photoSelected=>(prevItems) => [...prevItems,photoSelected.photo_id]);
        //  //console.log(toEditIds);
        //  return prevItems;
        // }

        const grouped = _.groupBy(sortedImagesUploaded, 'createdAt');
        const keys = Object.keys(grouped);
        const values = Object.values(grouped);
        const tempImgDated = [];

        useState(() => { for (let i = 0; i < keys.length; i += 1)
    {
      const imgCorresponding = values[i];
      // dateFormat("2019-04-30T08:59:00.000Z", "mmmm dS, yyyy")
      // tempImgDated.push(<div><h6>{monthName(keys[i])}</h6></div>);
      tempImgDated.push(<div><h6>{keys[i]}</h6></div>);
      tempImgDated.push(
        imgCorresponding.map((image) => (
            <ImagesCR
              key={image.id}
              Url={image.Url}
              image={image}
              onEdit={toggleModal}
              id={0}
            />
        )),
      );
      tempImgDated.push(<br />);
      setimgDated([...imgDated, ...tempImgDated]);
    } });
  // setModalIsOpen(true);

  return (

    <>

      <ul id="topNavbar" className="nav nav-tabs">
        <span className="col" />
        <div className="col-11">
          <li className="nav-item dropdown">
            <a className="nav-link dropdown-toggle text-secondary" data-bs-toggle="dropdown" href="/#" role="button" aria-expanded="false">Date uploaded</a>
            <div className="dropdown-menu" />
          </li>

        </div>
      </ul>
      {/* <SideNavBar /> */}
    <div className="sidephoto">
        <SideNavBar />
        <div className="row">
          <div className=" container_body">

            {/* {grouped.map((arr) => <h5>{arr[0]}</h5>)} */}
            {/* {map((arr) => ({ dateuploaded: arr[0], image: arr.slice(1) }))} */}
            { sortedImagesUploaded.map((image) => (
              <ImagesCR
                key={image.photo_id}
                url={image.photo_url}
                image={image}
                onEdit={toggleModal}
              />

            )) }
      {/* <div> */}
        {/* {imgDated} */}
      {/* </div> */}
      </div>

        </div>

      </div>
          <main className="main_Modal">
            {isModalOpen && (
            <Modal
              onRequestClose={closeMainModal}
              onEditRequest={toggleEditModal}
              onAddRequest={toggleAddModal}
              onDeleteRequest={toggleDelete}
              imgSelected={toEdit}
              countSelected={count}
              id={0}
            />
            )}
          </main>
      {/* </div> */}
      {/* <div className="container_body"> */}

      {/* </div> */}
      {/* <Modal /> */}

      <main className="main_edit">
        {isEditModalOpen && (
        <EditModal
          onRequestEditClose={toggleEditModal}
          imgEdit={toEdit}
          countEdit={count}
        />
        )}
      </main>
      <main className="main_edit">
        {isAddModalOpen && (
        <AddModal
          onRequestAddClose={toggleAddModal}
          onRequestCreate={toggleCreateAlbum}
          imgAdd={toEdit}
          countAdd={count}
          
        />
        )}
      </main>
      <main>
        {isDeleteOpen && (
        <DeleteModal
          onRequestClose={toggleDelete}
          onDelete={confirmDelete}
        />
        )}
      </main>



      <main>
        {isCreateAlbumOpen && (
        <CreateNewAlbumModal
          onRequestCreateClose={toggleCreateAlbum}
          imgIdsCreateAlbum={toEditIds}
          
        />
        )}
      </main>

    </>
  );
}

export default CamreRoll;
