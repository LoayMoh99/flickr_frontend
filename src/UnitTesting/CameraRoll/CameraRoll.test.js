import React from 'react';
import ReactDOM from 'react-dom';
import CamreRoll, {handleIncrement, handleDecrement, handleDelete, containsObject} from '../../components/CameraRoll/CamreRoll';
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })



test('the handleIncrement method increments count', () => {
  let c = handleIncrement(1)
  expect(c).toEqual(2)
})

test('the handleDecrement method decrements count', () => {
    let c = handleDecrement(2)
    expect(c).toEqual(1)
  })

  test('the handleDelete method deletes from array', () => {
    const array = [1,2,3]
    let arrReturned = handleDelete(3,array)
    expect(arrReturned).toEqual([3])
  })

  test('the containsObject method checks for an element in the array', () => {
      const list = [{_id:1},{_id:2}];
      const obj = {_id:1};
    let c = containsObject(obj,list);
    expect(c).toBeTruthy();
  })
  test('the containsObject method checks for an element is not in the array', () => {
    const list = [{_id:1},{_id:2}];
    const obj = {_id:3};
  let c = containsObject(obj,list);
  expect(c).toBeFalsy();
})
 
it('renders without crashing', ()=>{
    const div = document.createElement('div');
    ReactDOM.render(<CamreRoll />, div);

});

it('CameraRoll renders correctly enzyme', () => {
  const wrapper = shallow(<CamreRoll />)

  expect(toJson(wrapper)).toMatchSnapshot();
});