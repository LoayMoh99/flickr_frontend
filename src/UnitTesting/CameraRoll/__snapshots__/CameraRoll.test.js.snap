// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CameraRoll renders correctly enzyme 1`] = `
<Fragment>
  <ul
    className="nav nav-tabs"
    id="topNavbar"
  >
    <span
      className="col"
    />
    <div
      className="col-11"
    >
      <li
        className="nav-item dropdown"
      >
        <a
          aria-expanded="false"
          className="nav-link dropdown-toggle text-secondary"
          data-bs-toggle="dropdown"
          href="/#"
          role="button"
        >
          Date uploaded
        </a>
        <div
          className="dropdown-menu"
        />
      </li>
    </div>
  </ul>
  <div
    className="sidephoto"
  >
    <SideNavBar />
    <div
      className="row"
    >
      <div
        className=" container_body"
      />
    </div>
  </div>
  <main
    className="main_Modal"
  />
  <main
    className="main_edit"
  />
  <main
    className="main_edit"
  />
  <main />
  <main />
</Fragment>
`;
